pkg/types/validation/installconfig.go

```diff
@@ -108,12 +110,14 @@ func validateMachinePools(pools []types.MachinePool, fldPath *field.Path, platfo
 
 func validatePlatform(platform *types.Platform, fldPath *field.Path, openStackValidValuesFetcher openstackvalidation.ValidValuesFetcher) field.ErrorList {
 	allErrs := field.ErrorList{}
-	activePlatform := ""
+	activePlatform := platform.Name()
+	i := sort.SearchStrings(types.PlatformNames, activePlatform)
+	if i == len(types.PlatformNames) || types.PlatformNames[i] != activePlatform {
+		allErrs = append(allErrs, field.Invalid(fldPath, platform, fmt.Sprintf("platform must be one of: %s", strings.Join(types.PlatformNames, ", "))))
```

Comment:
This error does not make sense from the user's perspective. It is telling the user to specify a string value for the platform field, but the platform field does not take a string value.

Reply:
> This error does not make sense from the user's perspective...

I'm away from my keyboard today, but I've enabled maintainer edits if you want to push a fix.

Reply:
I've pushed 496b536 -> 432f283, rebasing onto master (so I can adjust the `TAGS` entry from #961) and adjusting this error message to recycle the wording that was already in master.